### ISSUE TITLE
Add more settings to remark_config 

### ISSUE THREAD

**Issue by BarryRufer** [umputun/remark42#472]
Hello.

Need more settings:
 - enable/disable markdown
 - enable/disable links
 - enable/disable rss
 - enable/disable preview
 - enable/disable date comment
 - enable/disable vote down

And how to activate anonymous set vote?

**Comment by umputun** [comment]
Need more context.

1. Why any of this needed? What the use case and how it makes remark42 better? 
1. What each of those settings mean? For example "disable markdown" means what exactly? Not to render markdown at all? Or not to allow enter markdown formatting?
1. How this supposed to work, i.e. if someone set "disable links" what will happen to all links he already had in the old comments? 
1. No clue what "date comment" is. Do you mean hide timestamp of the comment?

vote down already configurable, kind of, see README for `POSITIVE_SCORE` param.

> And how to activate anonymous set vote?

anonymous users can't comment and this is international as such votes make very little sense.

**Comment by BarryRufer** [comment]
> Why any of this needed? What the use case and how it makes remark42 better?

If there is more flexibility in the settings, of course it will be better.
> What each of those settings mean? For example "disable markdown" means what exactly? Not to render markdown at all? Or not to allow enter markdown formatting?

Only plain text. In order not to worry that they will insert spam links.

> How this supposed to work, i.e. if someone set "disable links" what will happen to all links he already had in the old comments?

```js
var remark_config = {
    markdown: 0,
    rss: 0,
    preview: 0,
    date: 0
}
```

I need users to comment more, rather than thinking about how to log in, how to make italic text, how it will look, etc.

![изображение](https://user-images.githubusercontent.com/57297188/68324084-5dca7100-00cf-11ea-817b-6f1c4cadcdae.png)

> No clue what "date comment" is. Do you mean hide timestamp of the comment?

Yes, remove the comment date, as in the screenshot above.

> anonymous users can't comment and this is international as such votes make very little sense.

It is very good that you care about such a problem, but let the site owner decide this. For example, I would like to give anonymous users the opportunity to comment and rate.

**Comment by umputun** [comment]
> more flexibility in the settings, of course, it will be better.

Well, this is not exactly accurate. Features set vs extra complexity of the code (and setup) is not a trivial tradeoff.

Based on your response the changes you have in mind are mostly UI related and you asking about hiding some UI elements. However the part " not to worry that they will insert spam links." is not really markdown related, but about the general rendering. 

To me it makes sense to combine all of this into a single parameter, smth like "simple_view" stipping all extras, like markdown, RSS, anything else and keeping a naked text box to enter a text comment.

However, I'm not sure what the goal of hiding comment date, can you provide some motivation for this?

>  but let the site owner decide this. For example, I would like to give anonymous users the opportunity to comment and rate.

See #473

**Comment by BarryRufer** [comment]
>However, I'm not sure what the goal of hiding comment date, can you provide some motivation for this?

I want the focus only on the form of comments, commentators and their comments. I don't want any extraneous text to distract from viewing.

Is this free software? Can I create my Docker image in which I will make the necessary changes to these files? I would remove everything superfluous and translate into 2 languages, Spanish and Russian.

![img](https://user-images.githubusercontent.com/57297188/68325864-f7dfe880-00d2-11ea-9709-c806877a9f26.png)

**Comment by umputun** [comment]
> Is this free software?

As free as it gets, see [LICENSE](https://github.com/umputun/remark/blob/master/LICENSE)

> Can I create my Docker image in which I will make the necessary changes to these files? I would remove everything superfluous and translate into 2 languages, Spanish and Russian.

Sure you can. However, you will have to maintain your own fork and reintegrate anything added to upstream if those fixes/features important to you. The same story for security fixes. For instance, you may have some troubles to merge smth you need, like #473 if your fork deviates from the original too much. Note: this is not enough to allow voting on JS side for anonymous unless the server-side adjusted to allow it.

**Comment by poohitan** [comment]
> Well, this is not exactly accurate. Features set vs extra complexity of the code (and setup) is not a trivial tradeoff.
I completely agree with @BarryRufer that the more possible configuration end user has, the better it is for him. Adding more settings won't make the setup any more complex, as there could always be default values. But if user want to configure something, he should be able to do this.

As for me UI configurations is a must, and not just as a "simple/rich mode". The more settings, the better.
If users don't have ability to use custom CSS, then UI should be configurable with settings. Even if custom CSS would be supported, UI settings would be still nice to have.

**Comment by paskal** [comment]
@BarryRufer #474 resolved via #502: you can set `simple-view` parameter in remark configuration to `true` and you'll have something like this:
<img width="670" alt="image" src="https://user-images.githubusercontent.com/712534/71840495-b7462080-30bd-11ea-91c0-e8cca2d6a4b1.png">
Please let us know if that's what you had in mind when you created this issue.

**Comment by paskal** [comment]
Also, anonymous voting enabled by `anon-vote` option (only together with `votes-ip`) and will be added in #473 shortly. If that's all, I would suggest you to subscribe to #473 and resolve this ticket.

**Comment by paskal** [comment]
@BarryRufer I believe all requested features are available in 1.5. Resolving this issue, feel free to reopen if something is still missing.
